### PR TITLE
Analytics email tab MAILPOET-5092

### PR DESCRIPTION
**Pull request by websupporter**
## Description
Implements the email tab as described in [MAILPOET-5092]

## Code review notes
Before this PR prices came already rendered from the API (e.g. https://github.com/mailpoet/mailpoet/blob/d740ef85d25fde6cb97bd8e911ef99879d066e91/mailpoet/lib/Automation/Integrations/MailPoet/Analytics/Controller/OverviewStatisticsController.php#L50C8-L53)

As we need to calculate revenue now in the React App, I introduced `@woocommerce/currency` and got rid of `revenue_rendered` from the API.

I use this PR also to get rid of the usage of `MailPoet\WooCommerce\Helper` and replace it with the integrations WooCommerce class in order to clean up a bit.

## QA notes

_N/A_

## Linked PRs

[MAILPOET-5092]

## Linked tickets

_N/A_

## After-merge notes

_N/A_

## Tasks

- [x] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [NA] I added sufficient test coverage (will add a general acc test at the end of this project)
- [x] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes

[MAILPOET-5092]: https://mailpoet.atlassian.net/browse/MAILPOET-5092?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MAILPOET-5092]: https://mailpoet.atlassian.net/browse/MAILPOET-5092?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/tabs/emails/index.tsx [hunk 1]
```diff
@@ -1,3 +1,121 @@
...
+  },
+  {
+    key: 'clicked',
+    label: __('Clicked', 'mailpoet'),
+    isLeftAligned: false,
+    isNumeric: true,
+  },
+  {
+    key: 'orders',
+    label: __('Orders', 'mailpoet'),
```
Comment (websupporter) [inline mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/tabs/emails/index.tsx:34]: Not sure. For context, do we use `woocommerce` here?

```
_x('Orders', 'woocommerce', 'mailpoet')
```

File: mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/tabs/emails/summary.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      }).format(data.opened),
+    },
+    {
+      label: __('clicked', 'mailpoet'),
+      value: Intl.NumberFormat(locale.toString(), {
+        notation: 'compact',
+      }).format(data.clicked),
+    },
+    {
+      label: __('orders', 'mailpoet'),
```
Comment (websupporter) [inline mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/tabs/emails/summary.tsx:47]: Is `woocommerce` the context we are using?

File: mailpoet/assets/css/src/components-automation-analytics/header.scss [hunk 3]
```diff
@@ -7,6 +7,10 @@
   .woocommerce-filters-filter {
     width: 100%;
 
+    .components-dropdown__content .components-popover__content {
+      padding: 0
+    }
+
```
Comment (websupporter) [inline mailpoet/assets/css/src/components-automation-analytics/header.scss:13]: Not really related to this PR. This aligns the date picker popover more to the appearance in WooCommerce itself.

**Comment by JanJakes** [comment]
@websupporter Could you pls rebase this? 🙏

**Comment by websupporter** [comment]
rebased @JanJakes

File: mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/tabs/emails/index.tsx (hunk 1 above)
Comment (JanJakes) [reply mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/tabs/emails/index.tsx:34]: @websupporter I don't think so. I also checked Woo and they use not context for "Orders" either.

File: mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/tabs/emails/summary.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    revenue: 0,
+  });
+
+  const summary =  [
+    { label: __('sent', 'mailpoet'), value: Intl.NumberFormat(locale.toString(), { notation: 'compact' }).format(data.sent) },
+    { label: __('opened', 'mailpoet'), value: Intl.NumberFormat(locale.toString(), { notation: 'compact' }).format(data.opened) },
+    { label: __('clicked', 'mailpoet'), value: Intl.NumberFormat(locale.toString(), { notation: 'compact' }).format(data.clicked) },
+    { label: __('orders', 'mailpoet'), value: Intl.NumberFormat(locale.toString(), { notation: 'compact' }).format(data.orders) },
+    { label: __('revenue', 'mailpoet'), value: formattedPrice(data.revenue) },
+    { label: __('unsubscribed', 'mailpoet'), value: Intl.NumberFormat(locale.toString(), { notation: 'compact' }).format(data.unsubscribed) },
```
Comment (JanJakes) [inline mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/tabs/emails/summary.tsx]: There's a lot of repetition, I think you could do:

```ts
const formatter = Intl.NumberFormat(locale.toString(), { notation: 'compact' });
formatter.format(...);
```

File: mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/store/actions/index.ts [hunk 5]
```diff
@@ -23,10 +23,22 @@ export function setSectionData(payload: Section) {
...
+    type: 'SET_SECTION_DATA',
+    payload,
+  };
+}
+
 export function* updateSection(
   section: Section,
   queryParam: Query | undefined = undefined,
 ) {
+  dispatch(storeName).resetSectionData(section);
```
Comment (JanJakes) [inline mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/store/actions/index.ts:41]: Nice loading placeholders 👍  Do you think we can put the number of emails somewhere and use that number as the number of placeholder rows?

File: mailpoet/lib/AdminPages/PageRenderer.php [hunk 6]
```diff
@@ -210,4 +211,18 @@ class_exists(Debugger::class)
       $notice->displayWPNotice();
     }
   }
+
+  private function woocommerceStoreConfig() {
```
Comment (JanJakes) [inline mailpoet/lib/AdminPages/PageRenderer.php]: Better would be with a verb (like `getWooCommerceStoreConfig()`)

File: mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/overview/index.tsx [hunk 7]
```diff
@@ -8,7 +8,7 @@ import { select, useSelect } from '@wordpress/data';
 import { MailPoet } from '../../../../../../mailpoet';
 import { OverviewSection, storeName } from '../../store';
 import { locale } from '../../../../../config';
-import {formattedPrice} from "../../formatter";
+import { formattedPrice } from '../../formatter';
```
Comment (JanJakes) [inline mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/overview/index.tsx:11]: Just curious, do you not use format-on-save?

**Review by JanJakes — COMMENTED** [review]
Looks mostly good, I noticed some issues in the UI:

--- 

1) The numbers and percentages don't seem to align for me very much:

Screenshot 2023-06-21 at 15 56 31

--- 

2) Zero clicks, but there's an "average" badge:

Screenshot 2023-06-21 at 15 57 01

--- 

3) Maybe some gap needed here? At the same time, looks like a bit too much padding from the right edge of the table:

Screenshot 2023-06-21 at 15 57 55

--- 

4) Orders, Revenue, and Unsubscribed have no subtitle. In Figma, there seem to be some:

Screenshot 2023-06-21 at 15 59 41

--- 

5) I see no badge for opens. Just double-checking, is that correct?

Screenshot 2023-06-21 at 16 01 59

--- 

6) The "sent", "opened", and other keywords in the footer seem to be gray in the designs:

Screenshot 2023-06-21 at 16 02 50

--- 

7) Should we order the emails same as they are in the automation?

Screenshot 2023-06-21 at 16 06 23

File: mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/tabs/emails/summary.tsx (hunk 4 above)
Comment (websupporter) [reply mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/tabs/emails/summary.tsx]: 26aacdbc7

File: mailpoet/lib/AdminPages/PageRenderer.php (hunk 6 above)
Comment (websupporter) [reply mailpoet/lib/AdminPages/PageRenderer.php]: Always the `get` :laughing: 

Fixed 585fa2857

File: mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/overview/index.tsx (hunk 7 above)
Comment (websupporter) [reply mailpoet/assets/js/src/automation/integrations/mailpoet/analytics/components/overview/index.tsx:11]: At some point I need your PHPStorm config. But usually, doesn't prettier-write run automatically. That was my impression. Just, sometimes, it seems it does not for some reason.

**Comment by websupporter** [comment]
> 1. The numbers and percentages don't seem to align for me very much:

1bf187921

> 2. Zero clicks, but there's an "average" badge

Hmmm... Right. In the search for how we rate, I stumpled upon this function for our statistic newsletters: https://github.com/mailpoet/mailpoet/blob/trunk/mailpoet/lib/Twig/Functions.php#L301-L309

1. There we would have the same problem
2. Not sure, how we would want to call it when no one or only very few person clicked. 

> 3. Maybe some gap needed here? At the same time, looks like a bit too much padding from the right edge of the table:

Good point: 6fd8e4734

> 4. Orders, Revenue, and Unsubscribed have no subtitle. In Figma, there seem to be some:

Had been discussed in Slack that we can remove it if the numbers are unclear. Do you understand immediately, what they mean? [link](https://a8c.slack.com/archives/C040TSG288Y/p1678098293049339)

> 5. I see no badge for opens. Just double-checking, is that correct?

In Figma I do not see a badge. I think this in intended.

> 6. The "sent", "opened", and other keywords in the footer seem to be gray in the designs:

I checked it against the tables in WooCommerce analytics. There they are black as well. Not sure, if we should deviate at this point.

> 7. Should we order the emails same as they are in the automation?

Good point. 7c7ce64dd

**Review by JanJakes — APPROVED** [review]
Thanks! 👏

**Review by Aschepikov — APPROVED** [review]
GJ!

**Pull request merged**